### PR TITLE
Enhancement: Enable Jetpack Compose

### PR DESCRIPTION
**Pull request by AbdhilahiRWabwire**
# Jetpack Compose

This pull request audits the Gradle Build, Updates Dependencies, Migrates from KAPT to KSP, Enables Jetpack Compose and Enables Dependabot

- dependabot.yml - Automated dependency updates built into GitHub
 
- https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
 
- https://github.com/dependabot

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
Is this about an existing bug or enhacement request? If yes would you mind providing the URL? Thanks!

**Comment by AbdhilahiRWabwire** [comment]
This is GitHub Dependabot. Dependabot automatically updates dependencies in GitHub Projects. Dependabot is built into GitHub.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - package-ecosystem: "gradle"
+    commit-message:
+      prefix: "Gradle"
+    directory: "/app"
+    schedule:
+      day: "saturday"
+      interval: "weekly"
+      time: "03:00"
+
+  - package-ecosystem: "maven"
```
Comment (nicolas-raoul) [inline .github/dependabot.yml:29]: Do we really need this? Do we use Maven dependencies?

Comment (AbdhilahiRWabwire) [reply .github/dependabot.yml:29]: Dependabot updates Gradle dependencies and Maven packages in unison. For this to work, both must be declared. Everyone loves dependabot.

**Comment by nicolas-raoul** [comment]
I think it would be great to have it indeed.
Would you mind trying that bot on your fork, so that we can see whether unit tests still pass after the bot has updated dependencies? There will probably be a lot of updates the first time, so better have the fixes ready. Thanks! :-)

**Comment by AbdhilahiRWabwire** [comment]
Understood

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Task Complete

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul AVD fails if JDK is updated but the current version of Android Studio uses JDK 17 and JBR 17

**Comment by nicolas-raoul** [comment]
Nice, that's very promising! :-)

Somehow the build fails with this error, any idea why?
`app/src/main/java/fr/free/nrw/commons/customselector/helper/OnSwipeTouchListener.kt: (42, 9): 'onFling' overrides nothing`

I am looking forward to testing the app with all of these updates. :-)

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Thank you sir. I am fixing that error now. It is com.android.tools.build.

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul 

```shell
OnSwipeTouchListener.kt: (42, 9): 'onFling' overrides nothing
```

This line of code requires a manual fix. I will have to go through the code base and fix it.

**Comment by nicolas-raoul** [comment]
Wow, thanks a lot! :-)
I believe that flings are only used in that activity.

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Problem Solved. Also, the window manager settings for the on swipe listener are deprecated. You may want to upgrade that at some point in the future.

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Fixed Accidental override: The following declarations have the same JVM signature (onFling(Landroid/view/MotionEvent;Landroid/view/MotionEvent;FF)Z):

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Execution failed for task ':app:connectedBetaDebugAndroidTest'.

**Comment by nicolas-raoul** [comment]
@AbdhilahiRWabwire I believe this pull request would be a wonderful addition to the app. If some updates are creating issues, how about using [ignore](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore) to leave these ones untouched for now? :-)

**Comment by AbdhilahiRWabwire** [comment]
@nicolas-raoul Fixed with dependency audit.

**Comment by AbdhilahiRWabwire** [comment]
Refactoring broken Kotlin files that use KAPT

**Comment by nicolas-raoul** [comment]
Thanks for all of the work!

You can mark your pull request as "draft" until it is ready to review. :-)

**Comment by AbdhilahiRWabwire** [comment]
I will mark this PR as a draft until it is ready to be reviewed

**Comment by AbdhilahiRWabwire** [comment]
GitHub is not granting me permission to convert this Pull Request into a Draft.

**Comment by AbdhilahiRWabwire** [comment]
Some Views require API 33 to function but I have upgraded to API 34. I can fix the compatibility issue in the Views by migrating them to Compose. This will take some time.

**Comment by AbdhilahiRWabwire** [comment]
I see the Jetpack Compose Migration has begun.

**Pull request closed without merging**